utils/github.py

```diff
@@ -24,6 +24,67 @@ def get_file_from_github(github_api_token, dependency_repo, path_to_file, dep_na
 
     return base64.b64decode(file_content).decode()
 
+def get_file_from_github_recursive(github_api_token, dependency_repo, path_to_file, dep_name):
+    path_to_file = path_to_file_without_dependency(path_to_file, dep_name)
+    user_slash_repo = parse_repo_link(dependency_repo['url'])
+    
+    direct_file_content = _get_direct_file(github_api_token, user_slash_repo, dependency_repo['relative_root'], path_to_file, dependency_repo['commit'])
+    if direct_file_content:
+        return direct_file_content
+    
+    return _recursive_search(github_api_token, user_slash_repo, dependency_repo['relative_root'], path_to_file, dependency_repo['commit'])
+
+
+def _get_direct_file(github_api_token, user_slash_repo, relative_root, path_to_file, commit):
+    github_api_url = f"https://api.github.com/repos/{user_slash_repo}/contents/{relative_root}/{path_to_file}?ref={commit}"
```

Comment:
maybe separate into a separate function these 2 lines of code to fetch from github api which common for `_get_direct_file` and `_recursive_searach`